src/extension/tab/tab.ts

```diff
@@ -1,30 +1,18 @@
 // This script is injected into each tab.
 import browser from "webextension-polyfill";
-import type { ClientMessage } from "../messages";
-import { createActor, createWindowActor } from "../actor";
 import {
+  createMessageBridge,
   createPortMessageAdapter,
   createWindowMessageAdapter,
 } from "../messageAdapters";
-import { createRPCBridge } from "../rpc";
 
 declare const __IS_FIREFOX__: boolean;
 
 const portAdapter = createPortMessageAdapter(() =>
   browser.runtime.connect({ name: "tab" })
 );
 
-const tab = createWindowActor<ClientMessage>(window);
-const devtools = createActor<ClientMessage>(portAdapter);
-
-createRPCBridge(portAdapter, createWindowMessageAdapter(window));
-
-devtools.forward("explorerSubscriptionTermination", tab);
-devtools.forward("explorerRequest", tab);
-
-tab.forward("registerClient", devtools);
-tab.forward("clientTerminated", devtools);
-tab.forward("explorerResponse", devtools);
+createMessageBridge(portAdapter, createWindowMessageAdapter(window));
```

Comment:
Lost in all the type refactoring is the biggest change which is the simplification of the message transfer between two actors. No more need for forwarding specific messages!